### PR TITLE
[MINOR] Create Swagger for Task and LogHours endpoints

### LINKED ISSUE #193: Create Swagger for Task and LogHours endpoints

### PR DESCRIPTION
**Pull request by bowsong**
closes #193 

@davidthai0387 can you check if all the loghours routes are correct

### CODE REVIEW

**Review by nasaku898 — APPROVED** [review]

File: backend/swagger/cirrusspark.yaml [hunk 1]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        - "application/json"
+      parameters:
+        - name: "email"
+          in: "path"
+          description: "Employee email"
+          required: true
+          type: "string"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1435]: Its not the caller email here

File: backend/swagger/cirrusspark.yaml [hunk 2]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+      - "application/json"
+      parameters:
+      - name: id
+        in: "path"
+        description: "Employee id"
+        required: true
+        type: "string"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1461]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 3]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        type: "string"
+      - in: "body"
+        name: "body"
+        description: "Updated pay"
+        required: true
+        schema:
+          $ref: ""
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1492]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 4]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+      - "application/json"
+      parameters:
+      - name: email
+        in: "path"
+        description: "Employee email"
+        required: true
+        type: "string"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1518]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 5]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1544]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 6]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        type: "string"
+      - in: "body"
+        name: "body"
+        description: "Updated hours by input type"
+        required: true
+        schema:
+          $ref: ""
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1575]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 7]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+      - "application/json"
+      parameters:
+      - name: email
+        in: "path"
+        description: "Employee email"
+        required: true
+        type: "string"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1600]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 8]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+      - "tasks"
+      summary: "Creates task"
+      description: ""
+      operationId: "createTask"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Created user object"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1621]: You mean task here?

File: backend/swagger/cirrusspark.yaml [hunk 9]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        required: true
+        schema:
+          $ref: ""
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1627]: Same error as before with the description

File: backend/swagger/cirrusspark.yaml [hunk 10]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+      tags:
+      - "tasks"
+      summary: "Gets all tasks"
+      description: ""
+      operationId: "getAllTask"
+      produces:
+      - "application/json"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1646]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 11]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        - "application/json"
+      parameters:
+      - name: taskId
+        in: "path"
+        description: "Task Id"
+        required: true
+        type: "string"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1672]: Same here

File: backend/swagger/cirrusspark.yaml [hunk 12]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        "404":
+          description: "Not found"
+        "401":
+          description: "Unauthorized"
+        "500":
+          description: "Unexpected error"
+        "200":
+          description: "Ok"
+    post:
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1681]: Is it a post here?

File: backend/swagger/cirrusspark.yaml [hunk 13]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        type: "string"
+      - in: "body"
+        name: "body"
+        description: "Updated task"
+        required: true
+        schema:
+          $ref: ""
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1703]: Same error with description

File: backend/swagger/cirrusspark.yaml [hunk 14]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        - "application/json"
+      parameters:
+      - name: taskId
+        in: "path"
+        description: "Task Id"
+        required: true
+        type: "string"
+      responses:
+        "400":
+          description: "Invalid caller email supplied"
```
Comment (Manraj45) [inline backend/swagger/cirrusspark.yaml:1728]: Same here

**Review by Manraj45 — CHANGES_REQUESTED** [review]
You have some minor errors. Please fix them.

File: backend/swagger/cirrusspark.yaml (hunk 12 above)
Comment (bowsong) [reply backend/swagger/cirrusspark.yaml:1679]: Yeah, but I didn't want to change the backend and break something this close to the release

File: backend/swagger/cirrusspark.yaml [hunk 15]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+          description: "Unauthorized"
+        "500":
+          description: "Unexpected error"
+        "200":
+          description: "OK"
+  /logHours/pay/{id}:
+    get:
+      tags:
+      - "loghours"
+      summary: "Get all pays belonging to {id}"
```
Comment (davidthai0387) [inline backend/swagger/cirrusspark.yaml:1448]: This route gets the pay with {id}, not all pays belonging to {id}.

File: backend/swagger/cirrusspark.yaml [hunk 16]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+        "401":
+          description: "Unauthorized"
+        "500":
+          description: "Unexpected error"
+        "200":
+          description: "OK"
+    put:
+      tags:
+      - "loghours"
+      summary: "Update pay belonging to {id}"
```
Comment (davidthai0387) [inline backend/swagger/cirrusspark.yaml:1473]: Same here, it's updating the pay identified by {id}.

File: backend/swagger/cirrusspark.yaml [hunk 17]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+          description: "Unauthorized"
+        "500":
+          description: "Unexpected error"
+        "200":
+          description: "Ok"
+  /logHours/pay/latest/{email}:
+    get:
+      tags:
+      - "loghours"
+      summary: "Get most recent pays belonging to {email}"
```
Comment (davidthai0387) [inline backend/swagger/cirrusspark.yaml:1505]: Singular pay

File: backend/swagger/cirrusspark.yaml [hunk 18]
```diff
@@ -1371,6 +1371,369 @@ paths:
...
+          description: "Unauthorized"
+        "500":
+          description: "Unexpected error"
+        "200":
+          description: "OK"
+  /logHours/inputType/{email}:
+    get:
+      tags:
+      - "loghours"
+      summary: "Get hours by input type belonging to {email}"
```
Comment (davidthai0387) [inline backend/swagger/cirrusspark.yaml:1531]: "Get hours input type" makes more sense here I think

**Review by davidthai0387 — CHANGES_REQUESTED** [review]
A few things to change but good work!
Side note: I missed Sam's PR for the call endpoints but some of the parameter names don't match the parameter in the path. This would result in semantic errors in some swagger editors. Not a big problem but we might want to open a task to fix those.

**Review by davidthai0387 — APPROVED** [review]

**Review by Manraj45 — APPROVED** [review]
Good job!

**Pull request merged**